lib/micro/case.rb

```diff
@@ -239,6 +239,19 @@ def Failure(type = :error, result: nil)
         __get_result(false, value, type)
       end
 
+
+      def Try(type = nil, catch: StandardError, on: {})
+        result_key = type || :try
+        block_value = yield
+        result = on[:success] || { result_key => block_value }
+
+        Success(type || :ok, result: result)
+      rescue *catch => e
+        result = on[:failure] || { result_key => e }
```

Comment:
@MatheusRich I'm not so sure about this. But, I think that we could use `:exception` as the default result_key in case of failure. The reason to do this is to keep the same output of the `Micro::Case::Safe` error handling. What is your opinion about this?

Reply:
I've use `:try` here to mirror what we did with `Check`. I could change to `:exception`, but wouldn't it be kinda strange since on success the default key is still `:try`?

Reply:
@MatheusRich because of what I said in this https://github.com/serradura/u-case/pull/114#discussion_r559254348. If the block returns a Hash, an exception will be the unique reason to happening a failure. So, in this case, have the same contract of the `Micro::Safe` will enforce a standard behavior for exception handling.